pkg/mfa/client.go

```diff
@@ -58,22 +58,32 @@ func (c *Client) init() {
 	}
 }
 
-// GetEnrollsOpts contains the options to create an Authentication Factor.
-type GetEnrollOpts struct {
+// Type represents the type of Authentication Factor
+type Type string
+
+// Constants that enumerate the available Types.
+const (
+	SMS  Type = "sms"
+	TOTP Type = "totp"
+)
+
+// EnrollFactorOpts contains the options to create an Authentication Factor.
+type EnrollFactorOpts struct {
+
 	// Type of factor to be enrolled (sms or totp).
-	Type string
+	Type Type
 
 	// Name of the Organization.
-	TotpIssuer string
+	TOTPIssuer string
 
 	// Email of user.
-	TotpUser string
+	TOTPUser string
 
 	// Phone Number of the User.
 	PhoneNumber string
 }
 
-type EnrollResponse struct {
+type AuthenticationFactor struct {
```

Comment:
I prefer `Factor` / `Challenge` since the package prefix complements the type meaning (`mfa.Factor` / `mfa.Challenge` )

There's some discussion from [Go docs](https://go.dev/doc/effective_go#package-names) around idiomatic naming in Go:

> For instance, the buffered reader type in the bufio package is called Reader, not BufReader, because users see it as bufio.Reader, which is a clear, concise name. Moreover, because imported entities are always addressed with their package name, bufio.Reader does not conflict with io.Reader. Similarly, the function to make new instances of ring.Ring—which is the definition of a constructor in Go—would normally be called NewRing, but since Ring is the only type exported by the package, and since the package is called ring, it's called just New, which clients of the package see as ring.New. Use the package structure to help you choose good names.

Reply:
Updating `AuthenticationFactor` to just `Factor` based on Jonatas' remarks